src/plugins/discover/public/application/apps/main/services/use_discover_state.ts

```diff
@@ -147,16 +147,22 @@ export function useDiscoverState({
    */
   const resetSavedSearch = useCallback(
     async (id?: string) => {
-      const newSavedSearch = await services.getSavedSearchById(id);
-      const newIndexPattern = newSavedSearch.searchSource.getField('index') || indexPattern;
-      newSavedSearch.searchSource.setField('index', newIndexPattern);
-      const newAppState = getStateDefaults({
-        config,
-        data,
-        savedSearch: newSavedSearch,
-      });
-      await stateContainer.replaceUrlAppState(newAppState);
-      setState(newAppState);
+      if (id) {
```

Comment:
This condition is redundant here. Try to change filters or query wilthout any saved search being selected. And after clicking `new`, search state (filters, query, selected fields) should be resetted. It happens due to [**this**](https://github.com/Dmitriynj/kibana/blob/56e17cd39cdfd31c294c2f27ccc1afe257e28ee9/src/plugins/discover/public/application/apps/main/services/use_url.ts#L26) reset call will lead to nothing.